### PR TITLE
[doc] improve xgboost4j-spark-gpu doc [skip ci]

### PR DESCRIPTION
**Pull request by wbo4958**

### CODE REVIEW

File: doc/jvm/xgboost4j_spark_gpu_tutorial.rst [hunk 1]
```diff
@@ -25,10 +25,10 @@ a dependency for your project. We provide both stable releases and snapshots.
 Data Preparation
 ================
 
-In this section, we use `Iris <https://archive.ics.uci.edu/ml/datasets/iris>`_ dataset as an example to
-showcase how we use Spark to transform raw dataset and make it fit to the data interface of XGBoost.
+In this section, we use the `Iris <https://archive.ics.uci.edu/ml/datasets/iris>`_ dataset as an example to
+showcase how we use Spark to transform a raw dataset and make it fit the data interface of XGBoost.
 
-Iris dataset is shipped in CSV format. Each instance contains 4 features, "sepal length", "sepal width",
+The iris dataset is shipped in CSV format. Each instance contains 4 features, "sepal length", "sepal width",
```
Comment (sameerz) [inline doc/jvm/xgboost4j_spark_gpu_tutorial.rst:31]:
```suggestion
The Iris dataset is shipped in CSV format. Each instance contains 4 features, "sepal length", "sepal width",
```

File: doc/jvm/xgboost4j_spark_gpu_tutorial.rst [hunk 2]
```diff
@@ -149,9 +149,8 @@ you can do it through setters in XGBoostClassifer:
 
 .. note::
 
-  In contrast to the XGBoost4J-Spark package, which needs to first assemble the numeric
-  feature columns into one column with VectorUDF type by VectorAssembler, the
-  XGBoost4J-Spark-GPU does not require such transformation, it accepts an array of feature
+  In different with XGBoost4j-Spark which accepts both a feature column with VectorUDT type and
+  an array of feature column names, XGBoost4j-Spark-GPU only accepts an array of feature
```
Comment (sameerz) [inline doc/jvm/xgboost4j_spark_gpu_tutorial.rst:153]: I think the original wording was better.

**Review by sameerz — CHANGES_REQUESTED** [review]

File: doc/jvm/xgboost4j_spark_gpu_tutorial.rst (hunk 2 above)
Comment (wbo4958) [reply doc/jvm/xgboost4j_spark_gpu_tutorial.rst:153]: @sameerz, right now, we have unified the setFeaturesCol(value: Array[String]) API for both CPU and GPU. So users do not need to do the features assembling for xgboost4j-spark.

File: doc/jvm/xgboost4j_spark_gpu_tutorial.rst [hunk 3]
```diff
@@ -149,9 +149,8 @@ you can do it through setters in XGBoostClassifer:
 
 .. note::
 
-  In contrast to the XGBoost4J-Spark package, which needs to first assemble the numeric
-  feature columns into one column with VectorUDF type by VectorAssembler, the
-  XGBoost4J-Spark-GPU does not require such transformation, it accepts an array of feature
+  In different with XGBoost4j-Spark which accepts both a feature column with VectorUDT type and
```
Comment (sameerz) [inline doc/jvm/xgboost4j_spark_gpu_tutorial.rst:152]:
```suggestion
  In contrast with XGBoost4j-Spark which accepts both a feature column with VectorUDT type and
```

**Review by sameerz — CHANGES_REQUESTED** [review]

**Review by trivialfis — APPROVED** [review]

**Pull request merged**